hooks/__tests__/useOnScreen.test.tsx

```diff
@@ -0,0 +1,81 @@
+import { type MutableRefObject, useRef } from 'react';
+import { render } from '@testing-library/react';
+import { useOnScreen } from '../useOnScreen';
+
+describe('useOnScreen', () => {
+  const intersectionObserverOriginal = window.IntersectionObserver;
+  const OnScreenRenderer = ({
```

Comment:
I feel we should use testing-library/hooks here rather than creating a test that renders a component with true/false, you might also be able to create a fake ref.

As we can't really test this functionality in the sense of having an actual screen. So it's all "fake", as Jest environment doesn't actually have a screen...